backend/de.metas.payment.sepa/base/src/main/java/de/metas/payment/sepa/api/impl/CreateSEPAExportFromPaySelectionCommand.java

```diff
@@ -90,35 +107,54 @@ public I_SEPA_Export run()
 
 	private I_SEPA_Export_Line createExportLine(@NonNull final I_C_PaySelectionLine line)
 	{
-		final I_C_Invoice sourceInvoice = line.getC_Invoice();
-		Check.assumeNotNull(line.getC_Invoice(), "Parameter line has a not-null C_Invoice; line={}", line);
+		final PaySelectionLineType lineType = paySelectionBL.extractType(line);
+		final String description;
+		final String structuredRemittanceInfo;
+		switch (lineType)
+		{
+			case Invoice ->
+			{
+				final I_C_Invoice sourceInvoice = invoiceBL.getById(InvoiceId.ofRepoId(line.getC_Invoice_ID()));
+				description = sourceInvoice.getDescription();
+				structuredRemittanceInfo = line.getReference();
+
+			}
+			case Refund ->
+			{
+				final PaymentId originalPaymentId = PaymentId.ofRepoId(line.getOriginal_Payment_ID());
+				final I_C_Payment originalPayment = paymentBL.getById(originalPaymentId);
+				description = originalPayment.getDescription();
+				structuredRemittanceInfo = msgBL.getTranslatableMsgText(MSG_SEPA_Payment_Refund).translate(Env.getADLanguageOrBaseLanguage());
+			}
+			default -> throw new AdempiereException(ERR_SEPA_Invoice_Nor_Payment_Set, line);
+		}
 
-		final I_C_BP_BankAccount bpBankAccount = InterfaceWrapperHelper.load(line.getC_BP_BankAccount_ID(), I_C_BP_BankAccount.class);
+		final BankAccount bpBankAccount =  bankAccountDAO.getById(BankAccountId.ofRepoId(line.getC_BP_BankAccount_ID()));
 
 		final I_SEPA_Export_Line exportLine = newInstance(I_SEPA_Export_Line.class, line);
 
 		exportLine.setAD_Org_ID(line.getAD_Org_ID());
 		exportLine.setAD_Table_ID(getTableId(I_C_PaySelectionLine.class));
 		exportLine.setRecord_ID(line.getC_PaySelectionLine_ID());
 		exportLine.setAmt(line.getPayAmt());
-		exportLine.setC_BP_BankAccount(bpBankAccount); // 07789: also setting the BP bank account so the following model validator(s) can more easily see evaluate what it is.
-		exportLine.setC_Currency_ID(bpBankAccount.getC_Currency_ID());
+		exportLine.setC_BP_BankAccount_ID(BankAccountId.toRepoId(bpBankAccount.getId()));
+		exportLine.setC_Currency_ID(CurrencyId.toRepoId(bpBankAccount.getCurrencyId()));
```

Comment:
getCurrencyId() returns not null... so u can call it getCurrencyId().getRepoId(), isn't it?